x-pack/plugins/triggers_actions_ui/common/utils.ts

```diff
@@ -5,4 +5,12 @@
  * 2.0.
  */
 
+import React, { forwardRef, ReactElement } from 'react';
+
 export const nonNullable = <T>(v: T): v is NonNullable<T> => v != null;
+
+export function typedForwardRef<T, P = {}>(
+  render: (props: P, ref: React.Ref<T>) => ReactElement
+): (props: P & React.RefAttributes<T>) => ReactElement {
+  return forwardRef(render) as any;
```

Comment:
nit: Does the cast to `any` affect the types of the component? Should we cast to something else?

Reply:
It's a trick to prevent `forwardRef` from losing the generic types, taken from [TotalTypescript](https://www.totaltypescript.com/forwardref-with-generic-components#:~:text=here's%20the%20new%20definition). I'm assuming the any is necessary to avoid errors, while the correct types are the surrounding ones